judge/views/api/api_v2.py

```diff
@@ -1,122 +1,593 @@
+        return {
+            'id': profile.id,
+            'username': profile.username,
+            'points': profile.points,
+            'performance_points': profile.performance_points,
+            'problem_count': profile.problem_count,
+            'rank': profile.display_rank,
+            'rating': profile.latest_rating,
+            'volatility': profile.latest_volatility,
+        }
+
+
+class APIUserDetail(APIDetailView):
+    model = Profile
+    slug_field = 'user__username'
+    slug_url_kwarg = 'user'
+
+    def get_object_data(self, profile):
+        solved_problems = list(
+            Submission.objects
+            .filter(
+                result='AC',
+                user=profile,
+                problem__is_public=True,
+                problem__is_organization_private=False,
+            )
+            .values('problem').distinct()
+            .values_list('problem__code', flat=True),
+        )
+
+        last_rating = profile.ratings.order_by('-contest__end_time').first()
+
+        contest_history = []
+        participations = (
+            ContestParticipation.objects
+            .filter(
+                user=profile,
+                virtual=ContestParticipation.LIVE,
+                contest__in=Contest.get_visible_contests(self.request.user),
+                contest__end_time__lt=self._now,
+            )
+            .order_by('contest__end_time')
+        )
+        for contest_key, score, cumtime, rating, volatility in participations.values_list(
+            'contest__key', 'score', 'cumtime', 'rating__rating', 'rating__volatility',
+        ):
+            contest_history.append({
+                'key': contest_key,
+                'score': score,
+                'cumulative_time': cumtime,
+                'rating': rating,
+                'volatility': volatility,
             })
 
-    resp['problems'] = {
-        'points': profile.points,
-        'solved': solved_problems,
-        'attempted': attempted_problems,
-        'authored': list(Problem.objects.filter(is_public=True, is_organization_private=False, authors=profile)
-                         .values_list('code', flat=True)),
-    }
+        return {
+            'id': profile.id,
+            'username': profile.user.username,
+            'points': profile.points,
+            'performance_points': profile.performance_points,
+            'problem_count': profile.problem_count,
+            'solved_problems': solved_problems,
+            'rank': profile.display_rank,
+            'rating': last_rating.rating if last_rating is not None else None,
+            'volatility': last_rating.volatility if last_rating is not None else None,
+            'organizations': list(profile.organizations.values_list('id', flat=True)),
+            'contests': contest_history,
+        }
+
+
+class APISubmissionList(APIListView):
+    model = Submission
+    basic_filters = (
+        ('user', 'user__user__username'),
```

Comment:
I feel like it's more efficient to run a query to lookup the profile than joining stuff into submissions here, but that's just a feeling.